staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -386,6 +389,9 @@ func (az *Cloud) getServiceLoadBalancerStatus(service *v1.Service, lb *network.L
 				if err != nil {
 					return nil, fmt.Errorf("get(%s): lb(%s) - failed to get LB PublicIPAddress Name from ID(%s)", serviceName, *lb.Name, *pipID)
 				}
+				if name, found := service.Annotations[ServiceAnnotationLoadBalancerPIPName]; found && name != "" {
```

Comment:
The changes here is required, let's rollback L392-394. This is because above `pipName, err := getLastSegment(*pipID)` has already got the correct public IP name.